API/Services/ReadingListService.cs

```diff
@@ -519,6 +519,14 @@ public async Task CreateReadingListsFromSeries(Series series, Library library)
         {
             _logger.LogWarning("There is a mismatch on StoryArc and StoryArcNumber for {FileName}. Def", filename);
         }
+        // ensure that both arrays are the same length
+        if (arcNumbers.Length < arcs.Length)
+        {
+            Array.Resize(ref arcNumbers, arcs.Length);
+        } else if (arcs.Length < arcNumbers.Length)
+        {
+            Array.Resize(ref arcs, arcNumbers.Length);
+        }
 
         var maxPairs = Math.Min(arcs.Length, arcNumbers.Length);
```

Comment:
The goal here is to ensure we only work up until the max number of pairs. This line is now completely useless. In addition, I'm not sure what your resizing is actually doing, as from a logic perspective, we calculate the min number of pairs then process them. You're doing the same, but incuring a sys-call to allocate a new array and the memory along with it, which adds up when this is called per file.

Reply:
I was attempted to honour the principle of the Math.Min statement, that was showing that there are circumstances where ArcNumbers might be larger then Arcs.